src/chapters/aspdotnet-model-classes/data-layer.rst

```diff
@@ -0,0 +1,284 @@
+.. _data-layer:
+
+Models and Data 
+===============
+
+In order to work with data, we need to add another element to our MVC application. Say for example,
+we want to do things like remove an event from our list. Well, if two events both have the same name, 
+how might we identify which of those items to delete? We can't yet. So we need to tweak how we store
+event data. 
+
+In ``CodingEvents``, we add a unique identifier field to ``Events`` to better handle and track distinct 
+``Event`` instances. We'll also create another model class called ``EventData`` to encapsulate data storage 
+and prepare ourselves for decoupling data from controller classes.
+
+Add a Unique Id - Video
+-----------------------
+
+.. TODO: Add create uid in model class video
+.. topics covered: create id field on event class
+
+YOUTUBE VIDEO HERE
+
+.. admonition:: Note
+
+   The starter code for this video is found at the `add-property branch <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/add-property>`__
+   of ``CodingEventsDemo``. The final code presented in this 
+   video is found on the `add-id branch <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/add-id>`__.
+   As always, code along to the videos on your own ``CodingEvents`` project.
+
+.. index:: ! uid, ! unique identifier
+
+Add a Unique Id - Text
+----------------------
+
+Identifying data by a user-defined string called ``name`` is not a sustainable or scalable method
+of handling data in most situations. Consider the address book example. How can
+we distinguish between two contact entries with the same name field? It is a frequent
```

Comment:
This secondary example does not make sense since you already gave an example earlier in the page. Maybe reference that to tie the page together. "As we saw above with our events list, we cannot know for sure which event to remove if two events have the same name."

Reply:
I see that this could feel repetitive but I see it more as a supplement - what about it isnt making sense for you?

Reply:
When I read through it, it feels a little bit like the question brought up in the intro goes unresolved. Whenever an author leaves a question unresolved and dives into another example, I find myself confused. However, if you pose the same question in the video and then resolve it there and this text is just providing a second example, I think that is fine!